app/presenters/openid_connect_user_info_presenter.rb

```diff
@@ -54,7 +54,8 @@ def uuid_from_sp_identity(identity)
   end
 
   def email_from_sp_identity
-    email_context.last_sign_in_email_address.email
+    EmailAddress.find(identity.email_address_id).email ||
+      email_context.last_sign_in_email_address.email
```

Comment:
`email_address_id` is going to be `nil` in most cases, right? For existing usage that is. We should probably avoid a database query in that case.

Maybe something like:

```suggestion
    if identity.email_address_id && email_address = EmailAddress.find(identity.email_address_id)
      email_address.email
    else
      email_context.last_sign_in_email_address.email
    end
```

Reply:
I'd be concerned about those errors. Did you dig into the specific tests to see why an email address record wouldn't exist at that point?

Reply:
I'd rather we fix the specs that are failing if they're not accurately reflecting the real-world expectations, rather than building tolerances into the runtime code.